sw/device/silicon_creator/rom_ext/rom_ext_start.S

```diff
@@ -208,14 +237,121 @@ _rom_ext_start_boot:
    */
   la gp, __global_pointer$
 
-  // Re-enable linker relaxation.
-  .option pop
-
   /**
-   * Jump to C Code
+   * Jump to C Code using `jr` to make sure we are jumping to the virtual address.
    */
   .extern rom_ext_main
-  tail rom_ext_main
+  la t0, rom_ext_main
+  jr t0
 
   // Set size so this function can be disassembled.
   .size _rom_ext_start_boot, .-_rom_ext_start_boot
+
+
+/**
+ * Configure the ibex simple address translation 
+ *
+ * @param a0 virtual address
+ * @param a1 size of the region
+ * @param a2 phisical address 
+ */
+  .globl _ibex_address_translation_set
+  .type _ibex_address_translation_set, @function
+_ibex_address_translation_set:
+
+  //mask = virtual_addr | ((size - 1) >> 1);
+  addi a1, a1, -1
+  srli a1, a1, 1
+  or a0, a0,a1
+
+  // Load ibex config base address
+  li t0, TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR
+
+  // Set ibus and dbus address matching 
+  sw a0, RV_CORE_IBEX_IBUS_ADDR_MATCHING_0_REG_OFFSET(t0)
+  sw a0, RV_CORE_IBEX_DBUS_ADDR_MATCHING_0_REG_OFFSET(t0)
+
+  // Set ibus and dbus address remap
+  sw a2, RV_CORE_IBEX_IBUS_REMAP_ADDR_0_REG_OFFSET(t0)
+  sw a2, RV_CORE_IBEX_DBUS_REMAP_ADDR_0_REG_OFFSET(t0)
+
+  // Enable address translation
+  li t1, 0x01
+  sw t1, RV_CORE_IBEX_IBUS_ADDR_EN_0_REG_OFFSET(t0)
+  sw t1, RV_CORE_IBEX_DBUS_ADDR_EN_0_REG_OFFSET(t0)
+ 
+ // Re-clobber all used registers.
+  li a0, 0x0
+  li a1, 0x0
+  li a2, 0x0
+  li t0, 0x0
+  li t1, 0x0
+  ret
+  // Set size so this function can be disassembled.
+  .size _ibex_address_translation_set, .-_ibex_address_translation_set
+
+
+#define EPMP_CFG_PMPCFG7_OFFSET (24)
```

Comment:
nit: I'd put this as a calculation so it is easier to update if someone needs to: `((7 * 8) % 32)`